cmd/keeper/keeper.go

```diff
@@ -1160,5 +1160,7 @@ func keeper(cmd *cobra.Command, args []string) {
 	}
 	go p.Start()
 
-	<-end
+	if err := <-end; err != nil {
```

Comment:
Receiving from a `nil` channel is an error. But here we're receiving an value from a `non-nil` channel. There's no restriction on the value you can send on a channel (it depends on the channel type) - if the channel's type is a reference type (e.g. pointers), then it will accept `nil` as a value - no problem.